lib/modules/manager/cnb/extract.ts

```diff
@@ -0,0 +1,95 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { getDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { type ProjectDescriptor, ProjectDescriptorToml } from './schema';
+
+const dockerPrefix = /^docker:\/\//;
+
+function isDockerRef(ref: string): boolean {
+  const schemaMatch = ref.match(/^([a-z0-9]+):\/\//);
+  if (schemaMatch && !schemaMatch[0].match(dockerPrefix)) {
```

Comment:
```suggestion
  if (schemaMatch && !ref.startsWith('docker://') {
```
string search if faster